modules/library/metadata/src/main/java/org/geotools/util/URLs.java

```diff
@@ -186,7 +186,12 @@ public static File urlToFile(URL url) {
             path3 = string.substring(simplePrefix.length() - 1);
         } else {
             String auth = url.getAuthority();
-            String path2 = url.getPath().replace("%20", " ");
+            String path2 = url.getPath();
+            try {
+                path2 = URLDecoder.decode(path2, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                throw new RuntimeException("Could not decode the URL to UTF-8 format", e);
```

Comment:
Throwing a runtime exception here seems a bit severe.
However, I see that that is what the method is already doing for a similar failure earlier, and that by the WC3 spec URL encoding is assumed to be UTF-8.

I am still a bit concerned this could cause problems, but it does seem to be OK.